lib/config.go

```diff
@@ -16,13 +16,17 @@ type ConfigMethods struct {
 	inst *Instance
 }
 
-// NewConfigMethods creates a configuration handle from an instance
-func NewConfigMethods(inst *Instance) *ConfigMethods {
-	return &ConfigMethods{inst: inst}
+// Name returns the name of this method group
+func (m *ConfigMethods) Name() string {
+	return "config"
 }
 
-// CoreRequestsName specifies this is a configuration handle
-func (m ConfigMethods) CoreRequestsName() string { return "config" }
+// ConfigMethods returns the `ConfigMethods` that the instance has registered
+// TODO (ramfox): as the dispatch refactoring nears completion, we should
+// re-evaluate this name
+func (inst *Instance) ConfigMethods() *ConfigMethods {
+	return &ConfigMethods{inst: inst}
+}
```

Comment:
getting caught up on this PR, I think we should take the other approach, calling _this_ config, and switching the existing `inst.Config()` method to `inst.GetConfig()` and marking `inst.GetConfig` as deprecated at the same time.

We're trying to get away from `*lib.Instance` having any non-method accessors, and this is a chance to move forward on that goal